You attempted to access the `__DESCRIPTOR__` property (of <frontend@service:prototypes::ember581>)

Related to #16148

I have a service, injected in a route (extending the AuthenticatedRouteMixin). When I do `this.get('prototypes')`,so, in my use case I don't know if the error is coming from: 
 * The route accessing a service
 * The route itself
 * The service itself

Let me now if you need more information.
For now, I will try to revert 2.18 (ember 3.0.X seems to have problems with ie 11)

```
There is more information at https://github.com/emberjs/ember.js/issues/16148, and
you can help us improve this error message by telling us more about what happened in
this situation.
    at new EmberError (error.js:40)
    at assert (index.js:139)
    at Object.get (core_object.js:57)
    at isDescriptorTrap (ember-metal.js:3258)
    at get (ember-metal.js:3430)
    at Class.get (observable.js:38)
    at route.js:62
```
cc/ @wycats 
Maybe this is an issue in ember-simple-auth, so kindly cc/ to @marcoow

```
ember-console.js:29 Error while processing route: widget-group.index Assertion Failed: You attempted to access the `__DESCRIPTOR__` property (of <frontend@service:prototypes::ember579>).
Since Ember 3.1, this is usually fine as you no longer need to use `.get()`
to access computed properties. However, in this case, the object in question
is a special kind of Ember object (a proxy). Therefore, it is still necessary
to use `.get('__DESCRIPTOR__')` in this case.

If you encountered this error because of third-party code that you don't control,
there is more information at https://github.com/emberjs/ember.js/issues/16148, and
you can help us improve this error message by telling us more about what happened in
this situation. Error: Assertion Failed: You attempted to access the `__DESCRIPTOR__` property (of <frontend@service:prototypes::ember579>).
Since Ember 3.1, this is usually fine as you no longer need to use `.get()`
to access computed properties. However, in this case, the object in question
is a special kind of Ember object (a proxy). Therefore, it is still necessary
to use `.get('__DESCRIPTOR__')` in this case.

If you encountered this error because of third-party code that you don't control,
there is more information at https://github.com/emberjs/ember.js/issues/16148, and
you can help us improve this error message by telling us more about what happened in
this situation.
    at new EmberError (http://localhost:4200/assets/vendor.js:25254:25)
    at assert (http://localhost:4200/assets/vendor.js:25497:15)
    at Object.get (http://localhost:4200/assets/vendor.js:50711:73)
    at isDescriptorTrap (http://localhost:4200/assets/vendor.js:37443:87)
    at get (http://localhost:4200/assets/vendor.js:37615:41)
    at Class.get (http://localhost:4200/assets/vendor.js:49512:34)
    at http://localhost:4200/assets/frontend.js:10830:18
    at tryCatcher (http://localhost:4200/assets/vendor.js:62499:21)
    at invokeCallback (http://localhost:4200/assets/vendor.js:62671:33)
    at publish (http://localhost:4200/assets/vendor.js:62657:9)
```

Concerning the involved properties It's hard to say. The error is thrown when I access the prototypes service from the afterModel hook of a route. Just `this.get('prototypes')` make the app crash.
Here is the code of the service if it helps (BTW Thanks to @kategengler for his addon).

```
import { camelize } from '@ember/string';
import { keys } from '@ember/polyfills';
import EmberObject from '@ember/object';
import Service from '@ember/service';

export default Service.extend({
  flags: EmberObject.create(),

  setup(flags) {
    var normalizedFlags = {};
    for (var flag in flags) {
      if (flags.hasOwnProperty(flag)) {
        normalizedFlags[this._normalizeFlag(flag)] = flags[flag];
      }
    }
    this.set('flags', EmberObject.create(normalizedFlags));
  },

  showAll() {
    return this.get('flags').getProperties(keys(this.get('flags')));
  },

  enable(flag) {
    var normalizedFlag = this._normalizeFlag(flag);
    this.get('flags').set(normalizedFlag, true);
    this.notifyPropertyChange(normalizedFlag);
  },

  enableAll() {
    keys(this.get('flags')).forEach(function(flag) {
      this.enable(flag);
    }, this);
  },

  disable(flag) {
    var normalizedFlag = this._normalizeFlag(flag);
    this.get('flags').set(normalizedFlag, false);
    this.notifyPropertyChange(normalizedFlag);
  },

  disableAll() {
    keys(this.get('flags')).forEach(function(flag) {
      this.disable(flag);
    }, this);
  },

  unknownProperty(key) {
    return this.enabled(key);
  },

  enabled(flag) {
    var normalizedFlag = this._normalizeFlag(flag);
    return !!this.get('flags').get(normalizedFlag);
  },

  _normalizeFlag(flag) {
    return camelize(flag);
  }
});

```

--- a/packages/ember-runtime/lib/system/core_object.js
+++ b/packages/ember-runtime/lib/system/core_object.js
@@ -98,6 +98,8 @@ function makeCtor() {
               property === 'willWatchProperty' ||
               property === 'didUnwatchProperty' ||
               property === 'didAddListener' ||
+              property === '__DESCRIPTOR__' ||
+              property === 'isDescriptor' ||
               property in target
             ) {
               return Reflect.get(target, property, receiver);

--- a/packages/ember-runtime/tests/system/core_object_test.js
+++ b/packages/ember-runtime/tests/system/core_object_test.js
@@ -1,3 +1,4 @@
+import { get } from 'ember-metal';
 import CoreObject from '../../system/core_object';
 
 QUnit.module('Ember.CoreObject');
@@ -35,4 +36,19 @@ QUnit.test('toString should be not be added as a property when calling toString(
   obj.toString();
 
   assert.notOk(obj.hasOwnProperty('toString'), 'Calling toString() should not create a toString class property');
+});
+
+QUnit.test('should not trigger proxy assertion when retrieving a proxy with (GH#16263)', function(assert) {
+  let someProxyishThing = CoreObject.extend({
+    unknownProperty() {
+      return true;
+    }
+  }).create();
+
+  let obj = new CoreObject({
+    someProxyishThing
+  });
+
+  let proxy = get(obj, 'someProxyishThing');
+  assert.equal(get(proxy, 'lolol'), true, 'should be able to get data from a proxy');
 });

